src/Parsers/Access/ASTCreateUserQuery.h

```diff
@@ -44,9 +44,8 @@ class ASTCreateUserQuery : public IAST, public ASTQueryWithOnCluster
     std::shared_ptr<ASTUserNamesWithHost> names;
     std::optional<String> new_name;
 
-    std::optional<AuthenticationData> auth_data;
-
-    mutable std::optional<String> temporary_password_for_checks;
+    mutable std::optional<AuthenticationData> auth_data;
+    mutable std::optional<String> temporary_password;
 
```

Comment:
It seems the code will be better if we create a new class like `AuthenticationData` but for use only in `AST`. For me it looks like it would be better if we create a class like `ASTAuthenticationData` which would be used in `ASTCreateUserQuery` only (not in `User`). And this `ASTAuthenticationData` would be convertible to `AuthenticationData` and back, and it would be different from it because it would be able to contain a plain password, as well as a password without specifying an authentication type.